dace/transformation/dataflow/redundant_array.py

```diff
@@ -1161,3 +1176,339 @@ def apply(self, sdfg: SDFG):
+                                                       popped)
+                # NOTE: This fixes the following case:
+                # A ----> A[subset] ----> ... -----> Tasklet
+                # Tasklet is not data, so it doesn't have an other subset.
+                if isinstance(e3.dst, nodes.AccessNode):
+                    e3.data.other_subset = other_subset
+                else:
+                    e3.data.other_subset = None
+                wcr = wcr or e3.data.wcr
+                wcr_nonatomic = wcr_nonatomic or e3.data.wcr_nonatomic
+                e3.data.wcr = wcr
+                e3.data.wcr_nonatomic = wcr_nonatomic
+
+            # Remove edge and add new one
+            graph.remove_edge(e2)
+            e2.data.wcr = wcr
+            e2.data.wcr_nonatomic = wcr_nonatomic
+            graph.add_edge(in_array, e2.src_conn, e2.dst, e2.dst_conn, e2.data)
+
+        # Finally, remove out_array node
+        graph.remove_node(out_array)
+        if out_array.data in sdfg.arrays:
+            try:
+                sdfg.remove_data(out_array.data)
+            except ValueError:  # Already in use (e.g., with Views)
+                pass
+
+
+@registry.autoregister_params(singlestate=True, strict=True)
+class RedundantWriteSlice(pm.Transformation):
+    """ Detects patterns of the form View(Array) -> Array and removes
+    the View if it is a slice. """
+
+    in_array = pm.PatternNode(nodes.AccessNode)
+    out_array = pm.PatternNode(nodes.AccessNode)
+
+    @staticmethod
+    def expressions():
+        return [
+            sdutil.node_path_graph(RedundantWriteSlice.in_array,
+                                   RedundantWriteSlice.out_array)
+        ]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        in_array = graph.nodes()[candidate[RedundantWriteSlice.in_array]]
+        out_array = graph.nodes()[candidate[RedundantWriteSlice.out_array]]
+
+        in_desc = in_array.desc(sdfg)
+        out_desc = out_array.desc(sdfg)
+
+        # Make sure that both arrays are using the same storage location.
+        if in_desc.storage != out_desc.storage:
+            return False
+
+        # The match must be View -> Array
+        if not (isinstance(out_desc, data.Array)
+                and not isinstance(out_desc, data.View)):
+            return False
+        if not isinstance(in_desc, data.View):
+            return False
+
+        # Ensure out degree is one (only one target, which is out_array)
+        if graph.out_degree(in_array) != 1:
+            return False
+
+        # The match must be View(Array) -> Array , i.e.,
+        # the View must point to the Array. Find the true in_desc.
+        true_in_array = sdutil.get_last_view_node(graph, in_array)
+        if not true_in_array:
+            return False
+        if true_in_array is not out_array:
+            return False
+        
+        # If the View receives data from a reduction, fail.
```

Comment:
:(

Reply:
The reduce node must be updated at some point I guess. The issue still exists.